src/sql/workbench/services/objectExplorer/browser/objectExplorerService.ts

```diff
@@ -407,67 +414,100 @@ export class ObjectExplorerService implements IObjectExplorerService {
-								const nodeType = node.nodeTypeId === NodeType.Folder ? node.label : node.nodeTypeId;
-								this._telemetryService.createErrorEvent(TelemetryKeys.TelemetryView.Shell, TelemetryKeys.TelemetryError.ObjectExplorerExpandError, undefined, errorType)
-									.withAdditionalProperties({
-										nodeType,
-										providerId: expandResult.providerId
-									}).send();
-							}
-
-						} else {
-							this.logService.error('OE provider returns empty result or providerId');
+					try {
+						let resultMap: Map<string, azdata.ObjectExplorerExpandInfo> = new Map<string, azdata.ObjectExplorerExpandInfo>();
+						let allProviders: azdata.ObjectExplorerProviderBase[] = [provider];
+
+						let nodeProviders = this._nodeProviders[providerId];
+						if (nodeProviders) {
+							nodeProviders = nodeProviders.sort((a, b) => a.group!.toLowerCase().localeCompare(b.group!.toLowerCase()));
+							allProviders.push(...nodeProviders);
 						}
+						this.logService.trace(`${session.sessionId}: got providers for node expansion: ${allProviders.map(p => p.providerId).join(', ')}`);
 
-						// When get all responses from all providers, merge results
-						if (resultMap.size === allProviders.length) {
+						const resolveExpansion = () => {
 							resolve(self.mergeResults(allProviders, resultMap, node.nodePath));
-
 							// Have to delete it after get all responses otherwise couldn't find session for not the first response
+							clearTimeout(timeout);
 							if (newRequest) {
 								delete self._sessions[session.sessionId!].nodes[node.nodePath];
 								this.logService.trace(`Deleted node ${node.nodePath} from session ${session.sessionId}`);
 							}
 						}
-					});
-					if (newRequest) {
-						allProviders.forEach(provider => {
-							self.callExpandOrRefreshFromProvider(provider, {
-								sessionId: session.sessionId!,
-								nodePath: node.nodePath,
-								securityToken: session.securityToken
-							}, refresh).then(isExpanding => {
-								if (!isExpanding) {
-									// The provider stated it's not going to expand the node, therefore do not need to track when merging results
-									let emptyResult: azdata.ObjectExplorerExpandInfo = {
-										errorMessage: undefined,
-										nodePath: node.nodePath,
-										nodes: [],
-										sessionId: session.sessionId
-									};
-									resultMap.set(provider.providerId, emptyResult);
+
+						// Incase node status not found.
+						this._onNodeExpandedError.event(e => {
+							resultMap.set(e.providerId, e);
+							// When get all responses from all providers, merge results
+							if (resultMap.size === allProviders.length) {
+								resolveExpansion();
+							}
+						});
+
+						const expansionTimeout = this._configurationService.getValue<number>('serverTree.nodeExpansionTimeout');
+						const timeout = setTimeout(() => {
```

Comment:
Would suggest having the timeout value be `expansionTimeoutValueSec` and the timeout be `expansionTimeout` so it's clear what the timeout is for